File: python-data-aggregator/services/shipping.py
from services.base import Base


class Shipping(Base):
    def get_data(self):
        return self.get("/")

    def get_quantity_on_hold(self, product_id):
        first_resp = [el for el in self.get_data() if el["id"] == product_id and el["status"] == "awaiting_shipping"][0]
        return first_resp['quantity']

File: data-aggregator/services/shipping.rb
require_relative 'base'

class Shipping < Base
  def initialize(base_url = "http://shipping")
    super(base_url)
  end

  def get_status_for_item(id)
    get_json("/").select{|item| item["id"] == id}.first["status"]
  end

  def get_quantity_on_hold(id)
    get_json("/").select{|item| item["id"] == id && item["status"] == "awaiting_shipping"}.first["quantity"]
  end
end